narwhal/primary/src/certificate_waiter.rs

```diff
@@ -149,7 +149,9 @@ impl CertificateWaiter {
                     waiting.push(fut).await;
                 }
                 // we poll the availability of a slot to send the result to the core simultaneously
-                (Some(certificate), permit) = try_fut_and_permit!(waiting.try_next(), self.tx_core) => {
+                Some((permit, Some(certificate))) = self.tx_core.with_permit(waiting.next()) => {
+                        // todo - consider if unwrap is a best action here
+                        let certificate = certificate.unwrap();
```

Comment:
maybe match on:
```
Some((permit, Some(Some(certificate))))
```

Although i'm not quite sure why they're is a double layer of Options

Reply:
So the unwrap below is for Result, not Option, I would prefer not to ignore the error

Reply:
Well previously the errors were ignored, so unwrapping here may cause an issue and crash the node. Perhaps the best thing to do now would be to log the error and then continue.

Reply:
If I read code of the macro correctly, it was logging the error and then panic?
```
                tracing::error!("{e}");
                panic!("I/O failure, killing the node.");
```

Reply:
But I think you are right in a sense that it seem more reasonable to log error here rather then panic